asf_search/Pair.py

```diff
@@ -0,0 +1,112 @@
+import importlib.util
+import math
+
+from .ASFProduct import ASFProduct
+from .baseline import calculate_perpendicular_baselines
+from .exceptions import CoherenceEstimationError
+import pytz
+
+_COHERENCE_OPT_DEPS = ['zarr', 's3fs', 'rioxarray', 'xarray']
+try:
+    for spec in _COHERENCE_OPT_DEPS:
+        if importlib.util.find_spec(spec) is None:
+            raise ImportError
+
+    import fsspec
+    import xarray as xr
+
+except ImportError:
+    fsspec = None
+    xr = None
+
+try:
+    from ciso8601 import parse_datetime
+except ImportError:
+    from dateutil.parser import parse as parse_datetime
+
+
+class Pair:
+    """
+    A Pair is comprised of a reference scene and a secondary scene. These scenes typically intersect geographically,
+    but that is not a requirement. When a pair is created, its perpendicular and temporal baselines are calculated
+    and stored in the self.perpendicular_baseline and self.temporal_baseline member variables.
+
+    Two pairs are equivalent if they have matching reference and secondary dates
+    """
+    def __init__(self, ref: ASFProduct, sec: ASFProduct):
+        self.ref = ref
+        self.sec = sec
+
+        self.perpendicular_baseline = calculate_perpendicular_baselines(
+            ref.properties['sceneName'], 
+            [sec, ref])[0].properties['perpendicularBaseline']
+
+        self.ref_time = parse_datetime(ref.properties["startTime"])
+        if self.ref_time.tzinfo is None:
+            self.ref_time = pytz.utc.localize(self.ref_time)
+        self.sec_time = parse_datetime(sec.properties["startTime"])
+        if self.sec_time.tzinfo is None:
+            self.sec_time = pytz.utc.localize(self.sec_time)
+
+        self.temporal_baseline = self.sec_time.date() - self.ref_time.date()
+
+    def __repr__(self) -> str:
+        return f"Pair({self.ref.properties['sceneName']}, {self.sec.properties['sceneName']})"
+
+    def __eq__(self, other):
+        if not isinstance(other, Pair):
+            return NotImplemented
+        return (self.ref.properties['sceneName'] == other.ref.properties['sceneName'] and
+                self.sec.properties['sceneName'] == other.sec.properties['sceneName'])
+
+    def __hash__(self) -> int:
+        return hash((self.ref.properties['sceneName'], self.sec.properties['sceneName']))
```

Comment:
For all three of these methods, you're essentially just using a tuple of the scene names, so you could add to the init method an `id` and use that here.

Something like:
```python
self.id = (ref.properties['sceneName'], sec.properties['sceneName'])
```

And then these methods just become:

```suggestion
    def __repr__(self) -> str:
        return f"Pair({self.id[0]}, {self.id[1]})"

    def __eq__(self, other):
        if not isinstance(other, Pair):
            return NotImplemented
        return self.id == other.id

    def __hash__(self) -> int:
        return hash(self.id)
```